### PR TITLE
Upstream SDK for Azure Web PubSub Service

### PR DESCRIPTION
**Pull request by terencefan**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/webpubsub/azure-webpubsub-upstream/azure/messaging/webpubsub/upstream/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    "InvalidEventTypeError",
+    "InvalidEventNameError",
+    "InvalidRequestError",
+    "HandlerNotFoundError",
+]
+
+from http import HTTPStatus
+
+
+class BaseError(Exception):
```
Comment (johanste) [inline sdk/webpubsub/azure-webpubsub-upstream/azure/messaging/webpubsub/upstream/exceptions.py:22]: This exception hierarchy needs to be re-evaluated. Most things should probably be `ValueError`s.

**Comment by iscai-msft** [comment]
hey @terencefan, thanks for opening a PR! Can you explain the purpose of `azure-webpubsub-upstream`? I also don't believe this SDK has gone to the archboard yet, and it would need to be to get approval for merge and release. Thanks!

File: sdk/webpubsub/azure-webpubsub-upstream/azure/messaging/webpubsub/upstream/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    @property
+    def event_name(self):
+        return self.context.event_name
+
+    def __init__(self, context):
+        super().__init__(context)
+
+
+class ValidationFailedError(ContextBasedError):
```
Comment (johanste) [inline sdk/webpubsub/azure-webpubsub-upstream/azure/messaging/webpubsub/upstream/exceptions.py:62]: Validation != authorization. The class name does not match the status code.

File: sdk/webpubsub/azure-webpubsub-upstream/setup.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    long_description=readme + '\n\n' + changelog,
+    long_description_content_type='text/markdown',
+    license='MIT License',
+    author='Microsoft Corporation',
+    author_email='azpysdkhelp@microsoft.com',
+    url='https://github.com/Azure/azure-sdk-for-python',
+    classifiers=[
+        'Development Status :: 4 - Beta',
+        'Programming Language :: Python',
+        'Programming Language :: Python :: 2',
```
Comment (johanste) [inline sdk/webpubsub/azure-webpubsub-upstream/setup.py:52]: No need to support py2 for azure libraries. Unless you have a compelling reason for your specific service, we recommend not adding support. Please note that libraries such as azure-core will no longer release new versions that support py2.

File: sdk/webpubsub/azure-webpubsub-upstream/setup.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    ],
+    zip_safe=False,
+    packages=find_packages(exclude=[
+        'tests',
+        # Exclude packages that will be covered by PEP420 or nspkg
+        'azure',
+    ]),
+    install_requires=[
+        'azure-common~=1.1',
+        'cloudevents==1.2.0'
```
Comment (johanste) [inline sdk/webpubsub/azure-webpubsub-upstream/setup.py:70]: Pinning a specific version of dependencies in setup.py will lead to version conflicts. Unless you have a very strong reason *for this specific dependency* to pin the version (this is exceedingly rare and something that would likely lead to us having to vendor the dependency), you should not do this.

File: sdk/webpubsub/azure-webpubsub-upstream/setup.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        'License :: OSI Approved :: MIT License',
+    ],
+    zip_safe=False,
+    packages=find_packages(exclude=[
+        'tests',
+        # Exclude packages that will be covered by PEP420 or nspkg
+        'azure',
+    ]),
+    install_requires=[
+        'azure-common~=1.1',
```
Comment (johanste) [inline sdk/webpubsub/azure-webpubsub-upstream/setup.py:69]: Do not pin the minor version.

Comment (lmazuel) [reply sdk/webpubsub/azure-webpubsub-upstream/setup.py:69]: You should not need this dependency at all actually, I didn't find any usage of it

File: sdk/webpubsub/azure-webpubsub-upstream/setup.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        'Development Status :: 4 - Beta',
+        'Programming Language :: Python',
+        'Programming Language :: Python :: 2',
+        'Programming Language :: Python :: 2.7',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.5',
+        'Programming Language :: Python :: 3.6',
+        'Programming Language :: Python :: 3.7',
+        'Programming Language :: Python :: 3.8',
+        'Programming Language :: Python :: 3.9',
```
Comment (lmazuel) [inline sdk/webpubsub/azure-webpubsub-upstream/setup.py:59]: 3.10 too:
https://github.com/Azure/azure-sdk-for-python/blob/main/doc/dev/packaging.md

File: sdk/webpubsub/azure-webpubsub-upstream/tests/test_exception.py [hunk 7]
```diff
@@ -0,0 +1,8 @@
+# coding=utf-8
```
Comment (lmazuel) [inline sdk/webpubsub/azure-webpubsub-upstream/tests/test_exception.py:1]: Delete sdk/webpubsub/azure-webpubsub-upstream/tests/__init__.py, a tests folder is not a package

File: sdk/webpubsub/azure-webpubsub-upstream/azure/messaging/webpubsub/upstream/utils.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
+# coding=utf-8
+# --------------------------------------------------------------------------
+# Created on Sun Sep 26 2021
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root
+# for license information.
+# --------------------------------------------------------------------------
+
+def parse_connection_string(connection_string, **kwargs):
```
Comment (lmazuel) [inline sdk/webpubsub/azure-webpubsub-upstream/azure/messaging/webpubsub/upstream/utils.py:10]: If no customer is supposed to use this file, prefix it with _

**Pull request closed without merging**